app/components/shared/PageWithContainer.js

```diff
@@ -1,15 +1,6 @@
-import React from 'react';
+import classed from './ClassedComponent';
 
-class PageWithContainer extends React.Component {
-  static propTypes = {
-    children: React.PropTypes.node.isRequired
-  };
-
-  render() {
-    return (
-      <div className="container">{this.props.children}</div>
-    );
-  }
-}
+const PageWithContainer = classed('div', 'container');
```

Comment:
This is an interesting idea, and `PageWithContainer` is an interesting use of it as well.

I've actually had the thought lately that some of these "dumb" components we have may be more hassle than they're worth. Let's take this example:

```
render() {
  return (
    <PageWithContainer>
      <div>Content 1</div>
      <div>Content 2</div>
    </PageWithContainer>
  )
}
```

We write this render function looks good. And "oh hey, there's this `PageWithContainer` that makes my life setting up a page with a container easy". I look at it in the browser, and looks good. We've abstracted away what `PageWithContainer` means, so if we want to change it globally, we just change it in one place.

In the future we decide to change `PageWithContainer` to have do `className="container flex"`, all of a sudden, every use of it is now broken. We've kinda broken what it means to be a `PageWithContainer`, so we'd need to make a new one `PageWithContainerFlex` or something..

I'd argue that `PageWithContainer` is actually worse than just putting `<div className="container">` in your render function:

```
render() {
  return (
    <div className="container">
      <div>Content 1</div>
      <div>Content 2</div>
    </div>
  )
}
```

It also means the next time we want to do something custom (like adding `flex`) we don't need to go adding new global components or things like that.

Unless it's a complex set of chained classNames (like Panel) for example, then I don't think it's worth having it abstracted like this! Also having dumb components means when you're onboarded with the codebase, you have to essentially learn an entire visual framework, instead of just using the tools you want how you want to.

And as the person that originally wrote `PageWithContainer` it was from my "Oh I heard about dumb components, it sounds good, we should use them" and not really knowing what a dumb component should be :P And I kinda regret adding it now.

When dealing with visual stuff and flex, you really need to know what the parent element is doing to make everything work. If you're writing a function and you see `PageWithContainer` you'd have to go look up the definition of that class to wonder "hey what does this actually do"...

Anyway, that was more of a rant at this class than at your approach! I think I'm just leaning more towards "dumb components that just have 1-2 class names should just be merged into the original code". I had this same problem when trying to fix something on the agents page. The `Panel` classes weren't working for me, so I had to go read how they work, and essentially copy/paste it's implementation into what I was doing, which kinda defeats the whole thing I think...